fastify.js

```diff
@@ -714,10 +717,13 @@ function fastify (options) {
           const id = genReqId(req)
           const childLogger = logger.child({ reqId: id })
 
-          childLogger.info({ req }, 'incoming request')
-
           const request = new Request(id, null, req, null, childLogger, onBadUrlContext)
           const reply = new Reply(res, request, childLogger)
+
+          if (disableRequestLogging === false) {
```

Comment:
I might be missing something, but what if this is set to `true` or `undefined`/`null` instead, it would ever be called?

Reply:
This is how it is checked in the other case when "incoming request" is logged, [in `route.js`](https://github.com/fastify/fastify/blob/109873cf7e63eeffae65de2613d50b21815d3cb3/lib/route.js#L471)

I suppose `if (disableRequestLogging === false) {...}` was used because `false` is the default value for this option.

However, IMHO the more generic `if (!disableRequestLogging) {...}` would be better. For example like in `reply.js` [here](https://github.com/fastify/fastify/blob/109873cf7e63eeffae65de2613d50b21815d3cb3/lib/reply.js#L606) and [here](https://github.com/fastify/fastify/blob/109873cf7e63eeffae65de2613d50b21815d3cb3/lib/reply.js#L779).

Reply:
>However, IMHO the more generic if (!disableRequestLogging) {...} would be better. For example like in reply.js [here](https://github.com/fastify/fastify/blob/109873cf7e63eeffae65de2613d50b21815d3cb3/lib/reply.js#L606) and [here](https://github.com/fastify/fastify/blob/109873cf7e63eeffae65de2613d50b21815d3cb3/lib/reply.js#L779).

Sure thing, that SGTM!

Nonetheless, I just mixed things a little and got confused between the name of the variable and the value itself. My bad 😅

Reply:
To summarize, any value, which is not exactly `false`, will disable this default logging. I'd suggest to keep this implementation, otherwise it may break someone's code (for example, if empty string is used as a value).